electrum/lnurl.py

```diff
@@ -9,23 +9,23 @@
 import urllib.parse
 
 import aiohttp.client_exceptions
-from aiohttp import ClientResponse
 
 from electrum import segwit_addr
 from electrum.segwit_addr import bech32_decode, Encoding, convertbits, bech32_encode
 from electrum.lnaddr import LnDecodeException, LnEncodeException
 from electrum.network import Network
 from electrum.logging import get_logger
 
-if TYPE_CHECKING:
-    from collections.abc import Coroutine
-
 
 _logger = get_logger(__name__)
 
 
 class LNURLError(Exception):
-    pass
+    def __init__(self, message="", *args):
+        # error messages are returned by the LNURL server, some services could try to trick
+        # users into doing something by sending a malicious error message
+        modified_message = f"[DO NOT TRUST THIS MESSAGE]:\n{message}"
+        super().__init__(modified_message, *args)
```

Comment:
if you are adding this prefix, you probably also want `util.error_text_str_to_safe_str()`

(but also, in many cases `message` is just a string literal from the source. Only in a few codepaths does it contain untrusted text. That suggests maybe the sanitisation should be moved closer to where the untrusted text is received)